ui/app/src/Views/AlertList/AlertView.elm

```diff
@@ -135,30 +164,41 @@ addLabelMsg ( key, value ) =
 
 silenceButton : Alert -> Html Msg
 silenceButton alert =
-    case alert.silenceId of
-        Just sId ->
-            a
-                [ class "btn btn-outline-danger border-0"
-                , href ("#/silences/" ++ sId)
-                ]
-                [ i [ class "fa fa-bell-slash mr-2" ] []
-                , text "Silenced"
-                ]
+    case alert.status of
+        Just status ->
+            case List.head status.silencedBy of
+                Just sId ->
+                    a
+                        [ class "btn btn-outline-danger border-0"
+                        , href ("#/silences/" ++ sId)
+                        ]
+                        [ i [ class "fa fa-bell-slash mr-2" ] []
+                        , text "Silenced"
+                        ]
+
+                Nothing ->
+                    a
+                        [ class "btn btn-outline-info border-0"
+                        , href (newSilenceFromAlertLabels alert.labels)
+                        ]
+                        [ i [ class "fa fa-bell-slash-o mr-2" ] []
+                        , text "Silence"
+                        ]
 
         Nothing ->
+            -- TODO: What to return here? This case should never happen.
```

Comment:
If this should never happen,  then why is status a maybe type?

Reply:
The definition of `Silence` is generated based on the new OpenAPI specification of the API v2. I would prefer not to have an additional internal type which converts to the generated one.

Why does the specification not enforce the `status` property: E.g. when creating a silence that status property is not yet known. Thereby the API does not enforce the property.

Is that enough context? What are your thoughts?

Reply:
I see, shouldn't a silence to be created and a silence that we retrieve be two different types?